controller/controller.go

```diff
@@ -248,75 +246,85 @@ func (c *ObjectStorageController) runWorker(ctx context.Context) {
 
 func (c *ObjectStorageController) processNextItem(ctx context.Context) bool {
 	// Wait until there is a new item in the working queue
-	op, quit := c.queue.Get()
+	uuidInterface, quit := c.queue.Get()
 	if quit {
 		return false
 	}
 
+	uuid := uuidInterface.(types.UID)
+	var err error
 	// With the lock below in place, we can safely tell the queue that we are done
 	// processing this item. The lock will ensure that multiple items of the same
 	// name and kind do not get processed simultaneously
-	defer c.queue.Done(op)
+	defer c.queue.Done(uuid)
 
 	// Ensure that multiple operations on different versions of the same object
 	// do not happen in parallel
-	c.OpLock(op)
-	defer c.OpUnlock(op)
+	c.OpLock(uuid)
+	defer c.OpUnlock(uuid)
+
+	op := c.opMap[uuid]
 
-	var err error
 	switch o := op.(type) {
 	case addOp:
 		add := *o.AddFunc
+		objMeta := o.Object.(metav1.Object)
+		name := objMeta.GetName()
+		ns := objMeta.GetNamespace()
 		err = add(ctx, o.Object)
+		if err == nil {
+			o.Indexer.Add(o.Object)
+		} else {
+			glog.Errorf("Error adding %s %s: %v", ns, name, err)
+		}
 	case updateOp:
 		update := *o.UpdateFunc
+		objMeta := o.OldObject.(metav1.Object)
+		name := objMeta.GetName()
+		ns := objMeta.GetNamespace()
 		err = update(ctx, o.OldObject, o.NewObject)
+		if err == nil {
+			o.Indexer.Update(o.NewObject)
+		} else {
+			glog.Errorf("Error updating %s %s: %v", ns, name, err)
+		}
 	case deleteOp:
 		delete := *o.DeleteFunc
+		objMeta := o.Object.(metav1.Object)
+		name := objMeta.GetName()
+		ns := objMeta.GetNamespace()
 		err = delete(ctx, o.Object)
+		if err == nil {
+			o.Indexer.Delete(o.Object)
+		} else {
+			glog.Errorf("Error deleting %s %s: %v", ns, name, err)
+		}
```

Comment:
in `default` block can we add the operator that was not handled, or is that logged earlier?

Reply:
sure